app/Http/Controllers/Api/UserActionApiController.php

```diff
@@ -0,0 +1,122 @@
+<?php namespace App\Http\Controllers\Api;
+/**
+ * Copyright 2023 OpenStack Foundation
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+use App\Http\Controllers\Api\OAuth2\OAuth2ProtectedController;
+use App\Http\Controllers\GetAllTrait;
+use App\ModelSerializers\SerializerRegistry;
+use Auth\Repositories\IUserActionRepository;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Request;
+use OAuth2\IResourceServerContext;
+use utils\Filter;
+use utils\FilterElement;
+use utils\FilterParser;
+use Utils\Services\ILogService;
+
+/**
+ * Class UserActionApiController
+ * @package App\Http\Controllers\Api
+ */
+final class UserActionApiController extends OAuth2ProtectedController
+{
+    use GetAllTrait;
+
+    /**
+     * UserActionApiController constructor.
+     * @param IUserActionRepository $repository
+     * @param IResourceServerContext $resource_server_context
+     * @param ILogService $log_service
+     */
+    public function __construct
+    (
+        IUserActionRepository  $repository,
+        IResourceServerContext $resource_server_context,
+        ILogService            $log_service
+    )
+    {
+        parent::__construct($resource_server_context, $log_service);
+        $this->repository = $repository;
+    }
+
+    /**
+     * @return array
+     */
+    protected function getFilterRules(): array
+    {
+        return [
+            'owner' => ['=='],
+            'realm' => ['=@', '=='],
+            'user_action' => ['=@', '=='],
+            'from_ip' => ['=@', '=='],
+            'created_at' => ['=='],
+        ];
+    }
+
+    /**
+     * @return array
+     */
+    protected function getFilterValidatorRules(): array
+    {
+        return [
+            'owner' => 'nullable|int',
```

Comment:
this filter should be owner_id and required

Reply:
done